tactics/class_tactics.ml

```diff
@@ -1305,9 +1303,9 @@ let resolve_all_evars depth unique env p oevd do_split fail =
           (* Typeclass resolution failed *)
           raise (Unresolved evd))
       with Unresolved evd' ->
-        if fail && (not do_split || is_mandatory (p evd') comp evd')
+        if fail && is_mandatory (p evd') comp evd'
         then (* Unable to satisfy the constraints. *)
-          let comp = if do_split then Some comp else None in
+          let comp = Some comp in
           error_unresolvable env comp evd'
```

Comment:
You should also adapt the type of `error_unresolvable` since it only takes a Some now.